### PR TITLE
[WIP] Trying to set up basic workflow

### PR DESCRIPTION
**Pull request by WackerO**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/differentialabundance/differentialabundance/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/differentialabundance.nf [hunk 1]
```diff
@@ -60,34 +63,54 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
...
     )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
-
-    CUSTOM_DUMPSOFTWAREVERSIONS (
-        ch_versions.unique().collectFile(name: 'collated_versions.yml')
+    ch_modified_samplesheet = IMPORTMERGEDCOUNTS.out.ch_modified_samplesheet //file(IMPORTMERGEDCOUNTS.out.ch_modified_samplesheet.first())
+    ch_modified_gene_counts = IMPORTMERGEDCOUNTS.out.ch_modified_gene_counts //file(IMPORTMERGEDCOUNTS.out.ch_modified_gene_counts.first())
+    ch_versions = ch_versions.mix(IMPORTMERGEDCOUNTS.out.versions)
+
+    ch_input = Channel.fromPath(params.contrast_meta)
```
Comment (pinin4fjords) [inline workflows/differentialabundance.nf:91]: If it helps, the way I imagined implementing this was that we'd have the input checker output the things that should 'travel together' as a channel (like in https://github.com/nf-core/modules/pull/2392/files, though appreciate that may not be the one we use).

Then we'd use a cross operation to make a channel with the matrix components plus the split-out contrasts. https://www.nextflow.io/docs/latest/operator.html#cross

Comment (pinin4fjords) [reply workflows/differentialabundance.nf:91]: There might be another operation or two involved to the above to get the right channel format.

Note that the current state of the DESeq2 module is to take a single channel: https://github.com/nf-core/modules/blob/master/modules/nf-core/deseq2/differential/main.nf

File: testdata/contrast_meta.csv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+id,variable,reference,target,blocking
```
[the remark this answers is not in the document]
Comment (ggabernet) [reply testdata/contrast_meta.csv:1]: you can also link to the modules branch if the data is already there

File: workflows/differentialabundance.nf [hunk 3]
```diff
@@ -60,34 +63,54 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
...
+        .splitCsv ( header:true, sep:',' )
+        .map{
+            tuple(
+                it,
+                ch_modified_samplesheet.val,
+                ch_modified_gene_counts.val
+                )
+        }
```
Comment (ggabernet) [inline workflows/differentialabundance.nf:99]:
```suggestion
        .map{
            meta_mapping (it)
        }
```

Comment (ggabernet) [reply workflows/differentialabundance.nf:99]: Here it's easier to define a function for mapping to the meta object that you need for the contrast. You can see examples of that here: https://github.com/nf-core/airrflow/blob/8c1185f7d659d7797a66b541bb27aa5861d34991/subworkflows/local/input_check.nf#L24

File: workflows/differentialabundance.nf [hunk 4]
```diff
@@ -60,34 +63,54 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
...
+
+    DESEQ2_DIFFERENTIAL (
+        ch_input,
+        IMPORTMERGEDCOUNTS.out.finished
```
Comment (ggabernet) [inline workflows/differentialabundance.nf:103]: ch_modified_samplesheet.collect(),
ch_modified_gene_counts.collect()

**Review by ggabernet — COMMENTED** [review]
Here some comments already, just noticeably, the `testdata` folder should not be there as the test data for pipelines should be hosted in the `github.com/nf-core/test-datasets` repo. If it's already available in the modules branch you can use that one. Also the modules tests are not included in the pipeline repo, these are only in the nf-core/modules repo.

File: workflows/differentialabundance.nf [hunk 5]
```diff
@@ -60,34 +63,54 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
...
+    ch_input = Channel.fromPath(params.contrast_meta)
+        .splitCsv ( header:true, sep:',' )
+        .map{
+            tuple(
+                it,
+                ch_modified_samplesheet.val,
+                ch_modified_gene_counts.val
+                )
+        }
+
```
Comment (ggabernet) [inline workflows/differentialabundance.nf:100]:
```suggestion

ch_for_deseq2 = ch_input.combine( ch_modified_samplesheet.collect()).combine(ch_modified_gene_counts.collect())

```

Comment (ggabernet) [reply workflows/differentialabundance.nf:100]: haven't tested this but something similar to this should work

**Comment by pinin4fjords** [comment]
Any chance of splitting this PR into smaller ones with specific objective to facilitate review? The interesting workflow changes are getting a bit lost among everything else.

File: workflows/differentialabundance.nf (hunk 1 above)
Comment (pinin4fjords) [reply workflows/differentialabundance.nf:91]: See also [my tweak to the DESeq2 module](https://github.com/pinin4fjords/nf-core-modules/blob/698d4a446dcd7f7302fa2d48994f6271a201f48b/tests/modules/nf-core/deseq2/differential/main.nf#L59) where I use `combine()` as @ggabernet was suggesting.

**Comment by WackerO** [comment]
@nf-core-bot fix linting

**Comment by WackerO** [comment]
@nf-core-bot fix linting

**Pull request closed without merging**